Upgraded to Poetry to Improve Dependency Management

codingwithsurya opened the pull request:
# Upgraded to Poetry to Improve Dependency Management + Code Cleanup
What user problem are we solving?
Currently, our Python backend strongly depends on Anaconda for environment and package management, resulting in a lack of separation between development and production environments. This creates challenges in maintaining and replicating the exact dependencies required for each environment.

What solution does this PR provide?

This pull request introduces the use of Poetry as a Python package manager tool to better define our development and production dependencies. By adopting Poetry, we can achieve better separation between the two environments and simplify the management of dependencies. In addition, this sped up build checks from 30mins to 5 mins. This migration also allows for venv caching as well.

The key changes in this PR include:

Created a pyproject.toml file in the poetry branch, which serves as the configuration file for Poetry.

Populated the pyproject.toml file with the necessary dependencies, both for development and production modes. The dependencies were inspired by the existing conda/environment.yml or requirements.txt files.

Updated the Python build check in the GitHub Actions workflow to use Poetry for environment setup. This ensures that the correct dependencies are installed during the CI/CD process.

Updated the Dockerfiles to use Poetry for environment setup. This ensures consistency between local development and deployment environments.

Ensured that installing Python dependencies updates the pyproject.toml file automatically. This eliminates the need to manually add dependencies to the conda/environment.yml file or requirements.txt.

**Code Cleanup**

Also some minor code cleanup occurred with the wiki.tsx file. The path names were wrong and directly needed the entire path when accessing the png file. So, import statements were deleted and path names were directly implemented in the code. Also the learn.tsx file was deleted as it was blank and not in use. 



Testing Methodology
How did you test your changes and verify that existing
functionality is not broken

To test these changes and verify that existing functionality is not broken, the following steps were taken:

Checked out the poetry-setup branch locally and ensured that Poetry was installed.

Built and ran the backend application locally using Poetry for environment setup. Confirmed that the application started successfully and all existing functionality was working as expected.

Triggered the CI/CD pipeline using the GitHub Actions workflow and verified that the build process completed without errors.

updated Docker files with the new Poetry-based environment setup and tested using docker build.

Requested feedback and conducted code reviews with team members to validate the changes and address any potential issues.

Any other considerations

Update documentation and ReadME files + we need to gitignore poetry.lock

add pre-commit hooks in poetry to make sure poetry.lock and pyproject.toml are correctly in sync

codingwithsurya committed:
updating poetry changes

codingwithsurya committed:
changing wiki.tsx with updated correct pathways

codingwithsurya commented:
no python-package-conda.yml action
no requirements.txt
no conda/environment.yml (conda deprecation)
update installation script in package.json at the root
update dockerfiles (train dockerfile + Dockerfile)
update readme to describe how to get setup on poetry (and add resource on pyenv to get the right python version)
add pre-commit hooks in poetry to make sure poetry.lock and pyproject.toml are correctly in sync
pre-commit hooks of interest to add are: poetry-check, poetry-lock. [Learn more](https://python-poetry.org/docs/master/pre-commit-hooks/)
How to build pre-commit hooks into github: [Here](https://pre-commit.com/)

Precommit YAML file to add to: [Here](https://github.com/DSGT-DLP/Deep-Learning-Playground/blob/nextjs/.pre-commit-config.yaml)


make sure to add logger to the list of packages as we just added it

codingwithsurya committed:
deleting learn.tsx and changing trainingcontainer.dockerfile

karkir0003 commented:
poetry-setup branch is not in use anymore. please update pr description? @codingwithsurya

karkir0003 commented:
you didnt actually deploy docker containers in your testing. all you did was build dockerfiles and run them with docker run <img name>

karkir0003 commented:
youll also want to explain the purpose of cleaning up imports in wiki.tsx and remove learn.tsx

maybe these import cleanups should be its own mini pr for better tracking!

codingwithsurya commented:
> youll also want to explain the purpose of cleaning up imports in wiki.tsx and remove learn.tsx
> 
> maybe these import cleanups should be its own mini pr for better tracking!

Let's just put this in the title and make it clear that we did some code cleanup for better tracking. Otherwise, as u saw before, i'm going to have to re-do all my changes to poetry branch. And honestly, the wiki.tsx file is the only one that needs to be explained, since learn.tsx will be implemented soon in learning modules and is currently blank.

karkir0003 commented:
so we are just lumping cleanup wirh poetry to save time? @codingwithsurya

codingwithsurya commented:
> so we are just lumping cleanup wirh poetry to save time? @codingwithsurya

yep. thats fine, right? we wanna get this out ASAP.

karkir0003 commented:
> > so we are just lumping cleanup wirh poetry to save time? @codingwithsurya
> 
> yep. thats fine, right? we wanna get this out ASAP.

considering the complexity of the extra cleanup, you're good. you can keep working on this PR

codingwithsurya committed:
adding pre-commit hooks, editing readme file, and editing trainingcontainerdockerfile

codingwithsurya committed:
Merge branch 'nextjs' into poetry

karkir0003 commented on .github/workflows/python-package-poetry.yml:
```diff
@@ -0,0 +1,61 @@
+        id: setup-python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9.16'
```
```suggestion
          python-version: '3.9.17'
```

should we use 3.9.17 to be consistent in our github actions @codingwithsurya

karkir0003 commented on .github/workflows/python-package-poetry.yml:
```diff
@@ -0,0 +1,61 @@
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
```
do we know if the latest version of poetry is installed within this action?

codingwithsurya replied:
yes. Current default settings for snok/install-poetry@v1 install poetry's latest version.

karkir0003 replied:
thanks for clarifying

karkir0003 commented on .gitignore:
```diff
@@ -165,6 +165,11 @@ backend/detection_results.csv
+#Poetry
+poetry.lock
+
+
```
remove extra blank newlines

karkir0003 commented on README.md:
```diff
@@ -58,15 +59,30 @@ If this command works properly, you will be redirected to an auth route in the G
+- Updating dependencies: `poetry update`
+
+# pyenv setup
+- To install pyenv, a python version management tool, you can use the following command via homebrew: `brew install pyenv`
```
i can add pyenv setup instructions for windows since the steps are a bit different

karkir0003 commented on README.md:
```diff
@@ -58,15 +59,30 @@ If this command works properly, you will be redirected to an auth route in the G
+
+- Verify the installation using `pyenv --version`
+
+- pyenv version control: `pyenv install -v 3.9.17`
```
what does this command do exactly? we should make it more clear to someone new to run commands 1, 2, 3 in step by step fashion

codingwithsurya replied:
I thought i gave it in a step by step format though, and at the end gave them command for pyenv version control and command if they run into any issues. Lmk how i can restructure it.

karkir0003 replied:
i think you should try to make it clear that for getting python 3.9.17 in pyenv, run the following commands. is the -v thing needed? what is the advantage you get?

codingwithsurya replied:
I was just showing the command for version control there for reference. I can remove it.

karkir0003 replied:
if its not very useful, then lets remove that part. i think brevity + clarity matter. a lot in a good readme.md

codingwithsurya replied:
good point

karkir0003 commented on frontend/src/pages/wiki.tsx:
```diff
@@ -180,9 +169,10 @@ const Wiki = () => {
               <p> Below is the dropout layer in action (notice the X marks) </p>
               <Image
-                src={dropout_dg}
+                src={"/images/wiki_images/dropout_diagram.png"}
```
should have constants for these types of paths towards the top of wiki.tsx  but right after the imports

it makes the code a bit more readable and its clear what these file paths signify

karkir0003 commented on frontend/src/pages/wiki.tsx:
```diff
@@ -256,9 +246,11 @@ const Wiki = () => {
 
               <Image
-                src={tanh_eq}
+                src={"/images/wiki_images/tanh_equation.png"}
```
same as above

karkir0003 commented on frontend/src/pages/wiki.tsx:
```diff
@@ -268,8 +260,10 @@ const Wiki = () => {
 
               <Image
-                src={tanh_plot}
+                src={"/images/wiki_images/tanh_plot.png"}
```
same as above

karkir0003 commented on frontend/src/pages/wiki.tsx:
```diff
@@ -310,8 +304,10 @@ const Wiki = () => {
               <p> Plot and Equation of the sigmoid function is shown below</p>
               <Image
-                src={sigmoid_eq}
+                src={"/images/wiki_images/sigmoid_equation.png"}
```
same as above

karkir0003 commented on frontend/src/pages/wiki.tsx:
```diff
@@ -359,9 +355,10 @@ const Wiki = () => {
 
               <Image
-                src={softmax_eq}
+                src={"/images/wiki_images/softmax_equation.png"}
```
same as above

karkir0003 commented on frontend/src/pages/wiki.tsx:
```diff
@@ -413,14 +410,18 @@ const Wiki = () => {
 
               <Image
-                src={conv2dgif}
+                src={"/images/wiki_images/conv2d.gif"}
```
same as above

karkir0003 commented on frontend/src/pages/wiki.tsx:
```diff
@@ -413,14 +410,18 @@ const Wiki = () => {
 
               <Image
-                src={conv2dgif2}
+                src={"/images/wiki_images/conv2d2.gif"}
```
same as above

karkir0003 commented on frontend/src/pages/wiki.tsx:
```diff
@@ -452,8 +453,10 @@ const Wiki = () => {
 
               <Image
-                src={maxpool2dgif}
+                src={"/images/wiki_images/maxpool2d.gif"}
```
same as above

karkir0003 commented on frontend/src/pages/wiki.tsx:
```diff
@@ -486,8 +489,10 @@ const Wiki = () => {
 
               <Image
-                src={avgmaxpoolgif}
+                src={"/images/wiki_images/avgpool_maxpool.gif"}
```
same as above

karkir0003 commented on frontend/src/pages/wiki.tsx:
```diff
@@ -528,8 +533,10 @@ const Wiki = () => {
 
               <Image
-                src={batchnorm}
+                src={"/images/wiki_images/batchnorm_diagram.png"}
```
same as above

karkir0003 commented on frontend/src/pages/wiki.tsx:
```diff
@@ -605,9 +612,11 @@ const Wiki = () => {
           <h2>Live Demo</h2>
           <Image
-            src={DEMO_VIDEO}
+            src={"/images/demo_video.gif"}
```
same as above

karkir0003 commented on pyproject.toml:
```diff
@@ -0,0 +1,51 @@
+
+[tool.poetry.dev-dependencies]
+python = ">=3.9"
+flake8 = "*"
```
we should make it clear that if you are developing, you should also have thr dev dependencies installed on top of the prod deps

codingwithsurya replied:
I see. so this means updating the ReadME file to provide clear instructions on installing both production and development dependencies?

maybe something like:

-  Install production dependencies: `poetry install --no-dev`
- Install development dependencies: `poetry install`

By installing both production and development dependencies, you will have all the necessary packages to run the application and also the additional tools for testing, linting, and code formatting.

karkir0003 replied:
yup. so poetry install = dev + prod deps? @codingwithsurya

codingwithsurya replied:
> . so poetry install = dev + prod deps?

yea exactly

karkir0003 reviewed (changes requested):
left some suggestions/questions @codingwithsurya

codingwithsurya committed:
Update .github/workflows/python-package-poetry.yml with correct python version

Co-authored-by: karkir0003 <54720987+karkir0003@users.noreply.github.com>

codingwithsurya committed:
code cleanup

- changing python version to stay consistent
- cleaning up whitespace in gitignore file
- Refactor wiki.tsx to improve code readability by defining constants for image paths near the top of the file

codingwithsurya committed:
Merge remote-tracking branch 'origin/poetry' into poetry

karkir0003 reviewed (changes requested):
responded to followups. i posted responses

codingwithsurya committed:
Changes in readme and dockerfile to manage dev/prod dependency

karkir0003 reviewed (commented):
looking good so far. @farisdurrani can u test poetry setup on your end?

plz address the followup

codingwithsurya committed:
deleting version control command from readme

karkir0003 reviewed (approved):
looks goid. conditionally approve based on @farisdurrani 's comments and @dwu359

karkir0003 commented:
@codingwithsurya I tested your poetry setup on my end. I got it to work for windows! backend and frontend spin up perfectly fine.

dwu359 commented on .gitignore:
```diff
@@ -165,6 +165,9 @@ backend/detection_results.csv
 backend/firebase_helpers/firebase_creds.json
 
+#Poetry
+poetry.lock
```
Commit the poetry.lock file: [https://python-poetry.org/docs/basic-usage/](https://python-poetry.org/docs/basic-usage/)

karkir0003 replied:
Thanks for catching this. @codingwithsurya can you do that?

karkir0003 replied:
I guess if we have `yarn.lock` in Version control, it shouldn't be suprising that `poetry.lock` should be there

karkir0003 commented:
@farisdurrani this PR can be merged contingent on the following
1. My [PR](https://github.com/DSGT-DLP/Deep-Learning-Playground/pull/867) gets merged into this branch
2. You test the poetry setup on your branch

dwu359 reviewed (commented):
I had to run the command `poetry env use $(pyenv which python)` to get poetry to use the version of python that I installed through pyenv to match the toml file python version requirements, anyone else?

karkir0003 commented:
> I had to run the command `poetry env use $(pyenv which python)` to get poetry to use the version of python that I installed through pyenv to match the toml file python version requirements, anyone else?

so you need to use the latest version of python 3.9 in order for `poetry install` to work @dwu359

dwu359 commented:
> > I had to run the command `poetry env use $(pyenv which python)` to get poetry to use the version of python that I installed through pyenv to match the toml file python version requirements, anyone else?
> 
> so you need to use the latest version of python 3.9 in order for `poetry install` to work @dwu359

So I ran
```
pyenv install 3.9
pyenv local 3.9
yarn startb
```
And that gave me an error unless I created a new poetry virtual env specifying the specific pyenv install for python 3.9

dwu359 commented:
I also ran into an error when building the docker image from Dockerfile: `Type error: '"@/features/Train/redux/trainspaceApi"' has no exported member named 'useGetColumnsFromDatasetQuery'. Did you mean 'useLazyGetColumnsFromDatasetQuery'?`
Someone probably forgot to change the names to match in a previous pr, shouldn't be hard to fix
This error was caused by the ImageParametersStep btw

dwu359 commented:
Would it be a good idea to have two separate poetry environments, one for the backend, and one for backendCore btw? We already have two separate docker files for each

karkir0003 commented:
> Would it be a good idea to have two separate poetry environments, one for the backend, and one for backendCore btw? We already have two separate docker files for each

that would be a good idea. if we know what deps are used on `backend/` vs. `backendCore`, we can make the builds probably more optimized?

farisdurrani commented:
@codingwithsurya can you please add `poetry shell` at the beginning of the `yarn run installb` instruction?

farisdurrani commented on README.md:
```diff
@@ -11,24 +11,25 @@ Web Application where people new to Machine Learning can input a dataset and exp
 | Action                                        | Command                                                                                    |
 | --------------------------------------------- | ------------------------------------------------------------------------------------------ |
 | Install (one-time) / Update Frontend Packages | `yarn run installf`                                                                        |
 | Install Backend Packages (one-time)           | `yarn run installb`                                                                        |
```
@codingwithsurya can you please add poetry shell at the beginning of the yarn run installb instruction?

karkir0003 replied:
I added this into `yarn run installb` @farisdurrani

farisdurrani commented on TrainingContainer.Dockerfile:
```diff
@@ -1,11 +1,30 @@
 FROM python:3.9-slim
```
@karkir0003 do you think we should instead use the full version of python instead of the slim version? I feel like it'll have more functions

karkir0003 replied:
But wouldn't the docker image then be a bit heavier? @farisdurrani. I think that the slim version anyway comes with pip

farisdurrani replied:
That is true. I guess if it works, it's fine

karkir0003 committed:
Pyenv Windows setup instructions (#867)

* windows setup instructions

* clarified need to pip install poetry

* removed whitespace

karkir0003 committed:
poetry lockfile

farisdurrani committed:
test

karkir0003 committed:
import fix

karkir0003 committed:
Merge branch 'poetry' of https://github.com/DSGT-DLP/Deep-Learning-Playground into poetry

karkir0003 committed:
poetry shell installb

karkir0003 commented:
> @codingwithsurya can you please add `poetry shell` at the beginning of the `yarn run installb` instruction?

added it in. Can you check @farisdurrani

karkir0003 commented:
> I also ran into an error when building the docker image from Dockerfile: `Type error: '"@/features/Train/redux/trainspaceApi"' has no exported member named 'useGetColumnsFromDatasetQuery'. Did you mean 'useLazyGetColumnsFromDatasetQuery'?` Someone probably forgot to change the names to match in a previous pr, shouldn't be hard to fix This error was caused by the ImageParametersStep btw



> I also ran into an error when building the docker image from Dockerfile: `Type error: '"@/features/Train/redux/trainspaceApi"' has no exported member named 'useGetColumnsFromDatasetQuery'. Did you mean 'useLazyGetColumnsFromDatasetQuery'?` Someone probably forgot to change the names to match in a previous pr, shouldn't be hard to fix This error was caused by the ImageParametersStep btw

updated that @dwu359

karkir0003 commented:
> @codingwithsurya can you please add `poetry shell` at the beginning of the `yarn run installb` instruction?

already added

karkir0003 committed:
poetry inits in two backend services

codingwithsurya committed:
different pyproject toml for aws-write and backend

codingwithsurya committed:
different pyproject toml for aws-write and backend

codingwithsurya committed:
:art: Format Python code with psf/black

karkir0003 committed:
updated github action venv path

codingwithsurya committed:
updated github actions

karkir0003 committed:
venv path run tests

karkir0003 committed:
Merge branch 'poetry' of https://github.com/DSGT-DLP/Deep-Learning-Playground into poetry

karkir0003 committed:
import fix

karkir0003 committed:
backend. import

karkir0003 committed:
backend. fix

karkir0003 committed:
s3 client import fix

karkir0003 committed:
import fix

karkir0003 committed:
moved tests into backend/ (#871)

* moved tests into backend/

* directory fix

karkir0003 committed:
venv fix

codingwithsurya commented:
as of saturday evening, everything works except docker builds/runs. The dockerfiles currently committed are a draft and need to be changed. Need some help on this. @karkir0003  @farisdurrani @dwu359 

also for some reason my docker builds take insanely long (15-20 mins) so need to figure out how to speed up the process.

codingwithsurya committed:
changing directory in aws_constants and DRAFT dockerfile changes

codingwithsurya committed:
:art: Format Python code with psf/black

karkir0003 committed:
Merge branch 'poetry' of https://github.com/DSGT-DLP/Deep-Learning-Playground into poetry

dwu359 commented:
> as of saturday evening, everything works except docker builds/runs. The dockerfiles currently committed are a draft and need to be changed. Need some help on this. @karkir0003 @farisdurrani @dwu359
> 
> also for some reason my docker builds take insanely long (15-20 mins) so need to figure out how to speed up the process.

What happens when you try to build a docker image?

karkir0003 commented:
> > as of saturday evening, everything works except docker builds/runs. The dockerfiles currently committed are a draft and need to be changed. Need some help on this. @karkir0003 @farisdurrani @dwu359
> > also for some reason my docker builds take insanely long (15-20 mins) so need to figure out how to speed up the process.
> 
> What happens when you try to build a docker image?

same. I tried making some updates/fixes to `Dockerfile`. The build takes pretty long, but the image size is like 12 GB on docker desktop. was wondering how we can optimize this. Thoughts @dwu359

codingwithsurya commented:
> > as of saturday evening, everything works except docker builds/runs. The dockerfiles currently committed are a draft and need to be changed. Need some help on this. @karkir0003 @farisdurrani @dwu359
> > also for some reason my docker builds take insanely long (15-20 mins) so need to figure out how to speed up the process.
> 
> What happens when you try to build a docker image?

So TrainingDockerFile built just fine but for the regular dockerfile we initially were facing errors with the path in `yarn run secrets` command. We fixed that and we were trying to re-test but dockerbuild was taking sooooo long (like around 20 mins or so). We were already 4-5 hours into working so at that point we were kinda tired and didn't keep trying to see if the test worked. I wonder if docker runs faster on your guys's machines.

karkir0003 commented:
> > > as of saturday evening, everything works except docker builds/runs. The dockerfiles currently committed are a draft and need to be changed. Need some help on this. @karkir0003 @farisdurrani @dwu359
> > > also for some reason my docker builds take insanely long (15-20 mins) so need to figure out how to speed up the process.
> > 
> > 
> > What happens when you try to build a docker image?
> 
> So TrainingDockerFile built just fine but for the regular dockerfile we initially were facing errors with the path in `yarn run secrets` command. We fixed that and we were trying to re-test but dockerbuild was taking sooooo long (like around 20 mins or so). We were already 4-5 hours into working so at that point we were kinda tired and didn't keep trying to see if the test worked. I wonder if docker runs faster on your guys's machines.

@codingwithsurya I was able to get `Dockerfile` to build on my machine. Had to make some minor fixes

karkir0003 commented:
I'm testing `docker build -t train -f TrainingContainer.Dockerfile . ...` @codingwithsurya

codingwithsurya commented:
> > > > as of saturday evening, everything works except docker builds/runs. The dockerfiles currently committed are a draft and need to be changed. Need some help on this. @karkir0003 @farisdurrani @dwu359
> > > > also for some reason my docker builds take insanely long (15-20 mins) so need to figure out how to speed up the process.
> > > 
> > > 
> > > What happens when you try to build a docker image?
> > 
> > 
> > So TrainingDockerFile built just fine but for the regular dockerfile we initially were facing errors with the path in `yarn run secrets` command. We fixed that and we were trying to re-test but dockerbuild was taking sooooo long (like around 20 mins or so). We were already 4-5 hours into working so at that point we were kinda tired and didn't keep trying to see if the test worked. I wonder if docker runs faster on your guys's machines.
> 
> @codingwithsurya I was able to get `Dockerfile` to build on my machine. Had to make some minor fixes

oh nice man! what were the changes?

karkir0003 commented:
lemme push it real quick. minor ordering of the commands

karkir0003 commented:
now the catch was that i didnt do a docker run. it took so long to see if it worked on my end.

codingwithsurya commented:
> now the catch was that i didnt do a docker run. it took so long to see if it worked on my end.

so both trainingdockerfile and dockerfile builds work now?

karkir0003 committed:
updated dockerfile ordering of commands

codingwithsurya commented:
> now the catch was that i didnt do a docker run. it took so long to see if it worked on my end.

yea no worries , it takes ages. docker run is pretty slow too. might just put a movie on to pass the time lol

karkir0003 commented:
@codingwithsurya pushed some of my changes/fixes for poetry

codingwithsurya commented:
> @codingwithsurya pushed some of my changes/fixes for poetry

awesome. good thought on switching to root directory right before COPY .. 

and all the consecutive commands cd to backend

karkir0003 commented:
> > @codingwithsurya pushed some of my changes/fixes for poetry
> 
> awesome. good thought on switching to root directory right before COPY ..
> 
> and all the consecutive commands cd to backend

can you give the docker build a try for `Dockerfile` and make sure the `docker run` works as intended. Also, im trying to test `docker build` for `TrainingContainer.Dockerfile`. Getting some frustrating connection pool timed out error from pypy when the `poetry install` is running

karkir0003 commented on backend/pyproject.toml:
```diff
@@ -0,0 +1,50 @@
+matplotlib = "*"
+seaborn = "*"
+mypy = "*"
+pytest = "*"
```
this should be a dev dependency

karkir0003 replied:
I'm referring to pytest

karkir0003 commented on backend/pyproject.toml:
```diff
@@ -0,0 +1,50 @@
+firebase_admin = "*"
+pyrebase4 = "*"
+moto = "*"
+docker = "*"
```
do we actually have a use case for `pip install docker` in our codebase. I wonder if this dependency is actually needed

karkir0003 commented on backend/pyproject.toml:
```diff
@@ -0,0 +1,50 @@
+pytest = "*"
+scikit-learn = "*"
+scipy = "1.7.3"
+pyyaml = "*"
```
not sure if this library is used anywhere in our codebase? @dwu359 do you have context?

dwu359 replied:
pytest seems like it's used for testing python code, scikit-learn and spicy are probably used for training, I'm not entirely sure what pyyaml is at first glance

karkir0003 replied:
Yeah. I think pytest is mainly for unit tests in python. I think it's mainly supposed to be a dev dependencies

codingwithsurya replied:
im going to resolve this for now in order to merge the pr, but let's address cleaning up dependencies in pyproject.toml in seperate pr

karkir0003 reviewed (commented):
left some comments/questions about what python deps we actually require

karkir0003 commented:
@codingwithsurya i got `TrainingContainer.Dockerfile` to build successfully.

karkir0003 commented:
@dwu359 potential tool for finding unused deps for our `pyproject.toml` files

https://github.com/fpgmaas/deptry

karkir0003 commented:
> @dwu359 potential tool for finding unused deps for our `pyproject.toml` files
> 
> https://github.com/fpgmaas/deptry

https://fpgmaas.github.io/deptry/rules-violations/

dwu359 commented:
> @dwu359 potential tool for finding unused deps for our `pyproject.toml` files
> 
> https://github.com/fpgmaas/deptry

Hmm yeah it does seem like something that would be neat to add to one of our GitHub workflows

karkir0003 commented:
> > @dwu359 potential tool for finding unused deps for our `pyproject.toml` files
> > https://github.com/fpgmaas/deptry
> 
> Hmm yeah it does seem like something that would be neat to add to one of our GitHub workflows

not sure if this library is good enough. I was trying to see if it has decent github star count + fairly recent commits to see if the repo is actively maintained

codingwithsurya committed:
Merge branch 'nextjs' into poetry
